src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -322,6 +473,312 @@ private int LookupAggregatorStore(KeyValuePair<string, object>[] tagKeysAndValue
+
+                                // Add to dictionary *after* initializing MetricPoint
+                                // as other threads can start writing to the
+                                // MetricPoint, if dictionary entry found.
+
+                                // Add the sorted order along with the given order of tags
+                                this.tagsToMetricPointIndexDictionaryDelta.TryAdd(sortedTags, lookupData);
+                                this.tagsToMetricPointIndexDictionaryDelta.TryAdd(givenTags, lookupData);
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    // This else block is for tag length = 1
+
+                    // Note: We are using storage from ThreadStatic, so need to make a deep copy for Dictionary storage.
+                    var givenTagKeysAndValues = new KeyValuePair<string, object>[length];
+
+                    tagKeysAndValues.CopyTo(givenTagKeysAndValues.AsSpan());
+
+                    givenTags = new Tags(givenTagKeysAndValues);
+
+                    lock (this.tagsToMetricPointIndexDictionaryDelta)
+                    {
+                        // check again after acquiring lock.
+                        if (!this.tagsToMetricPointIndexDictionaryDelta.TryGetValue(givenTags, out lookupData))
+                        {
+                            if (this.reclaimMetricPoints)
+                            {
+                                // Check for an available MetricPoint
+                                if (this.availableMetricPoints.Count > 0)
+                                {
+                                    index = this.availableMetricPoints.Dequeue();
+                                }
+                                else
+                                {
+                                    // No MetricPoint is available for reuse
+                                    return -1;
+                                }
+                            }
+                            else
+                            {
+                                index = ++this.metricPointIndex;
+                                if (index == this.metricPointReclamationThreshold)
+                                {
+                                    this.reclaimMetricPoints = true;
+                                }
+                            }
+
+                            lookupData = new LookupData(index, Tags.EmptyTags, givenTags);
+
+                            ref var metricPoint = ref this.metricPoints[index];
+                            metricPoint = new MetricPoint(this, this.aggType, givenTags.KeyValuePairs, this.histogramBounds, this.exponentialHistogramMaxSize, this.exponentialHistogramMaxScale, lookupData);
```

Comment:
I believe this line is relevant to @danelson's question during the SIG meeting today. In the event an exponential histogram point had been reclaimed, this line would ensure a new MetricPoint is instantiated with the configured max scale.

Is this correct?

Reply:
Just to be clear, I wasn't proposing one behavior was better/preferred over the other. I am not familiar with these internals or how backends might handle the data. I am simply raising the question in case it could yield any unexpected consequences.

Reply:
> I believe this line is relevant to @danelson's question during the SIG meeting today. In the event an exponential histogram point had been reclaimed, this line would ensure a new MetricPoint is instantiated with the configured max scale.
>
> Is this correct?

Yes.  That was a good point @danelson!

Would resetting the scale be acceptable? If not, we could selectively turn-off this behavior for exponential histograms if required. I'm not sure how backends which accept Delta would aggregate the measurements in such cases.

Reply:
I think it's ok for the scale to be reset here because the point may be reused by a different set of tags with different measurement characteristics.

That is, the range of measurements for a metric with tag A=1 may fit within a a higher scale than the same metric with tag A=2.